### PR TITLE
Add support for IPv6 for proxy address/hostname

### PR DESCRIPTION
**Pull request by accumulator**
Allows using proxy servers by IPv6 address, and proxy server hostnames that resolve to an IPv6 address.

Splits proxy authentication to --proxyuser and --proxypassword command line options and corresponding config keys, removing them from the single serialized proxy string.

### CODE REVIEW

File: electrum/network.py [hunk 1]
```diff
@@ -168,35 +168,52 @@ class NetworkParameters(NamedTuple):
...
-        n += 1
-    if len(args) > n:
-        proxy["password"] = args[n]
+    if args[0] in proxy_modes:
+        proxy['mode'] = args[0]
+        args = args[1:]
+
+    def is_valid_port(ps: str):
+        try:
+            assert 0 < int(ps) < 65535
```
Comment (SomberNight) [inline electrum/network.py:188]: I still think it's bad practice to rely on asserts being enabled.
just do:
```py
if not (0 < int(ps) < 65535):
   raise ValueError()
```

File: electrum/simple_config.py [hunk 2]
```diff
@@ -926,6 +926,8 @@ def _default_swapserver_url(self) -> str:
     NETWORK_AUTO_CONNECT = ConfigVar('auto_connect', default=True, type_=bool)
     NETWORK_ONESERVER = ConfigVar('oneserver', default=False, type_=bool)
     NETWORK_PROXY = ConfigVar('proxy', default=None)
+    NETWORK_PROXY_USER = ConfigVar('proxy_user', default=None)
+    NETWORK_PROXY_PASSWORD = ConfigVar('proxy_password', default=None)
```
Comment (SomberNight) [inline electrum/simple_config.py:930]: can we make these `type_=str`? I don't see why that would cause problems.

**Review by SomberNight — APPROVED** [review]
I haven't tested but at high-level it looks good.

**Comment by accumulator** [comment]
merged.

**Pull request closed without merging**